wagtail/contrib/modeladmin/templates/modeladmin/index.html

```diff
@@ -17,45 +17,30 @@
-                {% block header_extra %}
-                    <div class="right header-right">
-                        {% if view.list_export or user_can_create %}
-                            <div class="actionbutton">
-                                {% if user_can_create %}
-                                    {% include 'modeladmin/includes/button.html' with button=view.button_helper.add_button %}
-                                {% endif %}
-                                {% if view.list_export %}
-                                    <div class="dropdown dropdown-button match-width col">
-                                        <a href="?export=xlsx&{{ request.GET.urlencode }}" class="button bicolor button--icon">{% icon name="download" wrapped=1 %}{% trans 'Download XLSX' %}</a>
-                                        <div class="dropdown-toggle">{% icon name="arrow-down" %}</div>
-                                        <ul>
-                                            <li><a  class="button bicolor button--icon" href="?export=csv&{{ request.GET.urlencode }}">{% icon name="download" wrapped=1 %}{% trans 'Download CSV' %}</a></li>
-                                        </ul>
-                                    </div>
-                                {% endif %}
+        {% fragment as description %}
+            {% include 'modeladmin/includes/result_count.html' %}
+        {% endfragment %}
+        {% fragment as header_extra %}
+            <div class="right header-right">
+                {% if view.list_export or user_can_create %}
+                    <div class="actionbutton">
+                        {% if user_can_create %}
+                            {% include 'modeladmin/includes/button.html' with button=view.button_helper.add_button %}
+                        {% endif %}
+                        {% if view.list_export %}
+                            <div class="dropdown dropdown-button match-width col">
+                                <a href="?export=xlsx&{{ request.GET.urlencode }}" class="button bicolor button--icon">{% icon name="download" wrapped=1 %}{% trans 'Download XLSX' %}</a>
+                                <div class="dropdown-toggle">{% icon name="arrow-down" %}</div>
+                                <ul>
+                                    <li><a  class="button bicolor button--icon" href="?export=csv&{{ request.GET.urlencode }}">{% icon name="download" wrapped=1 %}{% trans 'Download CSV' %}</a></li>
+                                </ul>
                             </div>
                         {% endif %}
                     </div>
-                {% endblock %}
+                {% endif %}
             </div>
-            {% block extra_rows %}{% endblock %}
-        </header>
+        {% endfragment %}
+        {% include 'modeladmin/includes/header.html' with description=description icon=view.header_icon title=view.get_page_title subtitle=view.get_page_subtitle extra_actions=header_extra %}
```

Comment:
It would be great to avoid this additional include header - just for adding the search form.

An alternative approach could be something like this.

In `shared/header` - use a variable `search` (or similar, not `search_form` as that is used)
```html
            {% if search_url %}
                <form class="col search-form" action="{% url search_url %}{% if query_parameters %}?{{ query_parameters }}{% endif %}" method="get" novalidate role="search">
                    {% for field in search_form %}
                        {% include "wagtailadmin/shared/field.html" with field=field classname="w-mb-0" sr_only_label=True icon="search" %}
                    {% endfor %}
                    {% comment %}Add an initial disabled & hidden submit button so that pressing 'enter' will not submit form. Reload not required as the content is dynamically loaded.{% endcomment %}
                    <div class="visuallyhidden"><input disabled type="submit" aria-hidden="true"/></div>
                    <div class="submit visuallyhidden"><input type="submit" value="Search" class="button" /></div>
                </form>
            {% elif search %}
              {{ search }}
            {% endif %}
```

Then in the `modeladmin/index` use a fragment with a block inside.

```html
        {% fragment as search_fragment %}
            {% block search %}
                {% search_form %}
            {% endblock %}
        {% endfragment %}
        {% include 'wagtailadmin/shared/header.html' with description=description icon=view.header_icon title=view.get_page_title subtitle=view.get_page_subtitle extra_actions=extra_actions search=search_fragment %}
```
Note: we are using the shared/header include not the newly added file.